commons/src/main/java/com/powsybl/commons/datasource/ZipReadOnlyMemDataSource.java

```diff
@@ -0,0 +1,117 @@
+package com.powsybl.commons.datasource;
+
+import com.google.common.io.ByteStreams;
+import com.powsybl.commons.io.ForwardingInputStream;
+import org.apache.commons.compress.archivers.zip.ZipFile;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ZipReadOnlyMemDataSource implements ReadOnlyDataSource {
```

Comment:
Should we add this new class which is almost the same as ReadOnlyMemDataSource? The only difference is `putData`. I think we could have reused ReadOnlyMemDataSource in fact, and put the code you wrote in `putData` in `DataSourceUtil`, like the already existing one for file. Or we could have extended ReadOnlyMemDataSource maybe? what do you think?

Reply:
I actually agree, it should be possible to use only ReadOnlyMemDataSource. I will try to add the code in DataSourceUtil.